duo-client/src/main/java/com/duosecurity/client/canonicalization/Canonicalizer.java

```diff
@@ -0,0 +1,5 @@
+package com.duosecurity.client.canonicalization;
+
+public abstract class Canonicalizer {
+  public abstract String canonicalize(String method, String host, String url, String queryString);
```

Comment:
All c14 sig version requires method, host, url and queryString. Some additional info (like date, header, jsonbody) are required for different sig version. So I keep the same requirement in the method parameters and move others into constructor into individual version.